Streetcode/Streetcode.WebApi/Utils/WebParsingUtils.cs

```diff
@@ -126,92 +127,105 @@ public async Task ParseZipFileFromWeb(string mode)
-                System.Text.Encoding.RegisterProvider(System.Text.CodePagesEncodingProvider.Instance);
+            if (File.Exists(csvPath))
+            {
+                allLinesFromDataCSV = new List<string>(File.ReadAllLines(csvPath, Encoding.GetEncoding(1251)));
+                Console.OutputEncoding = Encoding.GetEncoding(1251);
+            }
 
-                string[] rows = File.ReadAllLines(excelPath, Encoding.GetEncoding(1251));
+            string excelPath = Directory.GetFiles(extractTo).First(fName => fName.EndsWith("houses.csv"));
 
-                // Grouping all rows from initial csv in order to get rid of duplicated streets
-                var groupedRows = rows
-                   .Select(x => x.Split(';'))
-                   .GroupBy(x => new { Column0 = x[0], Column1 = x[1], Column2 = x[2], Column3 = x[3], Column4 = x[4], Column6 = x[6] })
-                   .ToDictionary(g => g.Key, g => g.ToList());
+            List<string> rows = new List<string>(File.ReadAllLines(excelPath, Encoding.GetEncoding(1251)));
 
-                Console.ForegroundColor = ConsoleColor.Yellow;
-                Console.WriteLine("Import started. Rows to parse in original csv- " + rows.Length);
-                Console.WriteLine("Import started. Rows to parse in grouped csv- " + groupedRows.Count);
-                try
-                {
-                foreach (var row in groupedRows)
-                {
-                    if (Array.IndexOf(groupedRows.ToArray(), row) >= rowToStartReadingHousesCSV + 1 || (rowToStartReadingHousesCSV == 0))
-                    {
-                        var columns = row.Key;
-                        AddressEntity addressEntityBase = new AddressEntity() { Oblast = columns.Column0, AdminRegionOld = columns.Column1, AdminRegionNew = columns.Column2, Gromada = columns.Column3, City = columns.Column4, StreetName = columns.Column6 };
-                        string newRow = null;
+            // Grouping all rows from initial csv in order to get rid of duplicated streets
 
-                        // OSM Nominatim does not like e.g "м. Вінниця" - use "Вінниця" instead
-                        string cityStringSearchOptimized = addressEntityBase.City.Substring(addressEntityBase.City.IndexOf(" ") + 1);
+            var forParsindRows = rows
+                .Select(x => string.Join(";", x.Split(';').Take(7).ToList())).Distinct().ToList();
 
-                        string streetNameStringSearchOptimized = OptimizeStreetname(addressEntityBase.StreetName);
+            var alreadyParsedRows = allLinesFromDataCSV
+                .Select(x => string.Join(";", x.Split(';').Take(7).ToList())).Distinct().ToList();
 
-                        string addressrow = cityStringSearchOptimized + " " + streetNameStringSearchOptimized;
+            var alreadyParsedRowsToWrite = allLinesFromDataCSV
+                .Select(x => string.Join(";", x.Split(';').ToList())).Distinct().ToList();
```

Comment:
- From the context it is not entirely clear what thiscode does:
`.Select(x => string.Join(";", x.Split(';').Take(7).ToList())).Distinct().ToList()`.
Consider extracting this into the separate inline method, naming it according to its functionality.
- You don't need to enumerate the array returned from `.Split()` to a list, because `.Join()` works on IEnumerable interface.

Reply:
done

Reply:
Only the first part of the remark is resolved.

Reply:
done